packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java

```diff
@@ -90,6 +116,174 @@ private static BitmapDescriptor getBitmapFromBytes(List<?> data) {
+            return bitmapDescriptorFactory.fromBitmap(toScaledBitmap(bitmap, scale));
+          }
+        case "none":
+          break;
+      }
+      return bitmapDescriptorFactory.fromBitmap(bitmap);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Unable to interpret bytes as a valid image.", e);
+    }
+  }
+
+  /**
+   * Creates a BitmapDescriptor object from asset, using given details and density.
+   *
+   * <p>This method processes an asset specified by name and applies scaling based on the provided
+   * parameters. The `assetDetails` map must contain the keys 'assetName', 'bitmapScaling', and
+   * 'imagePixelRatio', and may optionally include 'width' and/or 'height' to explicitly set the
+   * dimensions of the output image.
+   *
+   * @param assetDetails a map containing the asset details and scaling instructions, with keys
+   *     'assetName': the name of the asset file, 'bitmapScaling': the scaling mode, either 'auto'
+   *     or 'none', 'imagePixelRatio': used with 'auto' scaling to compute the scale ratio, 'width'
+   *     (optional): the desired width, which affects scaling if 'height' is not provided, 'height'
+   *     (optional): the desired height, which affects scaling if 'width' is not provided
+   * @param assetManager assetManager An instance of Android's AssetManager, which provides access
+   *     to any raw asset files stored in the application's assets directory.
+   * @param density density the density of the display, used to calculate pixel dimensions.
+   * @param bitmapDescriptorFactory is an instance of the BitmapDescriptorFactoryWrapper.
+   * @param flutterInjector An instance of the FlutterInjectorWrapper class.
+   * @return BitmapDescriptor object from asset.
+   * @throws IllegalArgumentException if any required keys are missing in `assetDetails` or if the
+   *     asset cannot be opened or processed as a valid image.
+   */
+  @VisibleForTesting
+  public static BitmapDescriptor getBitmapFromAsset(
+      Map<?, ?> assetDetails,
+      AssetManager assetManager,
+      float density,
+      BitmapDescriptorFactoryWrapper bitmapDescriptorFactory,
+      FlutterInjectorWrapper flutterInjector) {
+
+    if (!assetDetails.containsKey("assetName")) {
+      throw new IllegalArgumentException("'asset' requires 'assetName' key.");
+    }
+    if (!assetDetails.containsKey("bitmapScaling")) {
+      throw new IllegalArgumentException("'asset' requires 'bitmapScaling' key.");
+    }
+    if (!assetDetails.containsKey("imagePixelRatio")) {
+      throw new IllegalArgumentException("'asset' requires 'imagePixelRatio' key.");
+    }
+
+    String assetName = toString(assetDetails.get("assetName"));
+    String assetKey;
+    try {
+      assetKey = flutterInjector.getLookupKeyForAsset(assetName);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("'asset' cannot open asset: " + assetName);
```

Comment:
Please include information from the original exception in the rethrown exception so that if it happens we can understand why and fix it.  
Something like `new IllegalArgumentException("'asset' cannot open asset: " + assetName, e);`